aci/resource_aci_fvnsmcastaddrinstp.go

```diff
@@ -0,0 +1,307 @@
+							Required: true,
+						},
+					}, GetBaseAttrSchema(), GetNameAliasAttrSchema()),
+				},
+			},
+		}, GetBaseAttrSchema(), GetNameAliasAttrSchema()),
+	}
+}
+
+func getRemoteMulticastAddressPool(client *client.Client, dn string) (*models.MulticastAddressPool, error) {
+	fvnsMcastAddrInstPCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	fvnsMcastAddrInstP := models.MulticastAddressPoolFromContainer(fvnsMcastAddrInstPCont)
+	if fvnsMcastAddrInstP.DistinguishedName == "" {
+		return nil, fmt.Errorf("MulticastAddressPool %s not found", dn)
+	}
+	return fvnsMcastAddrInstP, nil
+}
+
+func setMulticastAddressPoolAttributes(fvnsMcastAddrInstP *models.MulticastAddressPool, d *schema.ResourceData) (*schema.ResourceData, error) {
+	d.SetId(fvnsMcastAddrInstP.DistinguishedName)
+	d.Set("description", fvnsMcastAddrInstP.Description)
+	fvnsMcastAddrInstPMap, err := fvnsMcastAddrInstP.ToMap()
+	if err != nil {
+		return d, err
+	}
+
+	d.Set("annotation", fvnsMcastAddrInstPMap["annotation"])
+	d.Set("name", fvnsMcastAddrInstPMap["name"])
+	d.Set("name_alias", fvnsMcastAddrInstPMap["nameAlias"])
+	return d, nil
+}
+
+func getMulticastAddressBlocks(callType, multicastAddressPool string, client *client.Client, d *schema.ResourceData) []*models.MulticastAddressBlock {
+	log.Printf("[DEBUG] Beginning GET called by %s function for address pool name %s", callType, multicastAddressPool)
+	readMulticastAddressBlockData, err := client.ListMulticastAddressBlock(multicastAddressPool)
+	if err == nil {
+		log.Printf("[DEBUG] Finished GET called by %s successfully with result: %v", callType, d.Get("multicast_address_block"))
+	} else {
+		log.Printf("[DEBUG] Error during GET operation of multicast address blocks: %v", err)
+		readMulticastAddressBlockData = nil
+	}
+	return readMulticastAddressBlockData
+}
+
+func setMulticastAddressBlocks(callType, multicastAddressPool string, readMulticastAddressBlockData []*models.MulticastAddressBlock, client *client.Client, d *schema.ResourceData) {
+	log.Printf("[DEBUG] Beginning SET called by %s function for address blocks %v", callType, readMulticastAddressBlockData)
+	multicastAddressBlockList := make([]interface{}, 0)
+	for _, record := range readMulticastAddressBlockData {
+		multicastAddressBlockMap := make(map[string]interface{})
+		multicastAddressBlockMap["annotation"] = record.Annotation
+		multicastAddressBlockMap["from"] = record.From
+		multicastAddressBlockMap["name"] = record.Name
+		multicastAddressBlockMap["name_alias"] = record.NameAlias
+		multicastAddressBlockMap["to"] = record.To
+		multicastAddressBlockMap["dn"] = fmt.Sprintf(models.DnfvnsMcastAddrBlk, multicastAddressPool, record.From, record.To)
+		multicastAddressBlockList = append(multicastAddressBlockList, multicastAddressBlockMap)
+	}
+	d.Set("multicast_address_block", multicastAddressBlockList)
+	log.Printf("[DEBUG] Finished SET called by %s successfully with result: %v", callType, d.Get("multicast_address_block"))
```

Comment:
why use d.Get("multicast_address_block") when multicastAddressBlockList should be the same value?

Reply:
correct will change this